src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,134 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+
+    private static final int INITIAL_CAPACITY = 10;
+
+    private Object[] objects;
+    private int size;
+
+    public ArrayList() {
+        this.objects = new Object[INITIAL_CAPACITY];
+    }
+
     @Override
     public void add(T value) {
-
+        if (size() == objects.length) {
+            grow();
+        }
+        objects[size()] = value;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (index > size() || index < 0) {
+            throw new ArrayListIndexOutOfBoundsException("Index is invalid!");
+        }
+        if (index == size() - 1 && index == size()) {
+            this.add(value);
+        } else {
+            shift(index);
+            objects[index] = value;
+            size++;
+        }
```

Comment:
```suggestion
        System.arraycopy(objects, index, objects, index + 1, size - index);
        objects[index] = value;
        size++;
```
this is simplier

Reply:
I got the error

java.lang.ArrayIndexOutOfBoundsException: arraycopy: last destination index 11 out of bounds for object array[10]